### PR TITLE
commands: Added new option `cleanDestinationDirKeepDotFolder`

### PR DESCRIPTION
**Pull request by eamonnmcevoy**
Add a new option to ignore 'dot' folders when deleting files in destination dir.

Fixes #3208

### CODE REVIEW

**Comment by eamonnmcevoy** [comment]
relates to: https://github.com/spf13/hugo/issues/3202

File: commands/hugo.go [hunk 1]
```diff
@@ -578,9 +580,16 @@ func (c *commandeer) copyStatic() error {
...
 	// We can effectively clean the publishDir on initial sync
-	syncer.Delete = c.Cfg.GetBool("cleanDestinationDir")
+	syncer.Delete = c.Cfg.GetBool("cleanDestinationDir") || c.Cfg.GetBool("cleanDestinationDirKeepDotFolders")
+
 	if syncer.Delete {
 		c.Logger.INFO.Println("removing all files from destination that don't exist in static dirs")
+
+		if c.Cfg.GetBool("cleanDestinationDirKeepDotFolders") {
+			syncer.DeleteFilter = func(f os.FileInfo) bool {
+				return f.IsDir() && len(f.Name()) > 0 && string(f.Name()[0]) == "."
```
Comment (bep) [inline commands/hugo.go]: Use strings.HasPrefix(f.Name(), ".")

File: docs/content/commands/hugo_benchmark.md [hunk 2]
```diff
@@ -21,37 +21,38 @@ hugo benchmark
 ### Options
 
 ```
```
Comment (bep) [inline docs/content/commands/hugo_benchmark.md]: All doc below /commands is autogenerated so no need to edit it by hand.

File: commands/hugo.go [hunk 3]
```diff
@@ -578,9 +580,16 @@ func (c *commandeer) copyStatic() error {
 	syncer.DestFs = c.Fs.Destination
 	// Now that we are using a unionFs for the static directories
 	// We can effectively clean the publishDir on initial sync
-	syncer.Delete = c.Cfg.GetBool("cleanDestinationDir")
+	syncer.Delete = c.Cfg.GetBool("cleanDestinationDir") || c.Cfg.GetBool("cleanDestinationDirKeepDotFolders")
```
Comment (bep) [inline commands/hugo.go]: We have enough flag. 

I would say that having a cleanDestinationDir command that also deleted hidden folders ("." files) was a surprise for many. So I suggest we just make this the new default and make sure to document it. People who want to remove the entire folder can use `rm -rf`.

Comment (eamonnmcevoy) [inline commands/hugo.go]: No problem, I'm not sure where to update the documentation, do I just update `commands/hugo.go`?

ln:215:
`cmd.Flags().Bool("cleanDestinationDir", false, "Remove files from destination not found in static directories. Folders beginning with "." are ignored")`

**Pull request merged**